arm_diags/src/diurnal_cycle.py

```diff
@@ -102,87 +142,131 @@ def diurnal_cycle_data(parameter):
 
 
+#=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 def func24(x,p1,p2):
      return p1*np.sin(2*np.pi/24*x+p2)
 
 def diurnal_cycle_plot(parameter):
     """Calculate diurnal cycle climatology"""
+    sites=parameter.sites
     variables = parameter.variables
     seasons = parameter.season
     output_path = parameter.output_path
     ref_models = parameter.ref_models
 
     var_longname = [ varid_longname[x] for x in variables]
     mod_num = len(ref_models)
-    
+
+    # create site-dependent folder [XZ]
+    # Generate new folder given site names:
+    if not os.path.exists(os.path.join(output_path,'figures',sites[0])):
+        os.makedirs(os.path.join(output_path,'figures',sites[0])) 
+
+    # start the plotting procedure
     for j, variable in enumerate(variables):
         for i, season in enumerate(seasons):
-            test_data = genfromtxt(output_path+'/metrics/'+variable+'_'+season+'_test_diurnal_cycle.csv')
-            mmm_data =  genfromtxt(output_path+'/metrics/'+variable+'_'+season+'_mmm_diurnal_cycle.csv')
-            obs_data =  genfromtxt(output_path+'/metrics/'+variable+'_'+season+'_obs_diurnal_cycle.csv')
-            cmip_data = genfromtxt(output_path+'/metrics/'+variable+'_'+season+'_cmip_diurnal_cycle.csv')
+            # check the test model data
+            test_findex = 0 #preset of test model indicator
+            test_file = glob.glob(output_path+'/metrics/'+sites[0]+'/'+variable+'_'+season+'_test_diurnal_cycle_'+sites[0]+'.csv')
+            if len(test_file) == 0:
+               print('No test model plotted for diurnal cycle: '+sites[0])
+            if len(test_file) > 0: 
+                test_findex = 1   #test model exist
+            # read data
+            if test_findex == 1: test_data = genfromtxt(output_path+'/metrics/'+sites[0]+'/'+variable+'_'+season+'_test_diurnal_cycle_'+sites[0]+'.csv')
+            mmm_data =  genfromtxt(output_path+'/metrics/'+sites[0]+'/'+variable+'_'+season+'_mmm_diurnal_cycle_'+sites[0]+'.csv')
+            obs_data =  genfromtxt(output_path+'/metrics/'+sites[0]+'/'+variable+'_'+season+'_obs_diurnal_cycle_'+sites[0]+'.csv')
+            cmip_data = genfromtxt(output_path+'/metrics/'+sites[0]+'/'+variable+'_'+season+'_cmip_diurnal_cycle_'+sites[0]+'.csv')
             mod_num = cmip_data.shape[0]
-    
+
+            # plotting
             fig = plt.figure()# Create figure
-            ax  =fig.add_axes([0.15, 0.14, 0.8, 0.8]) # Create axes
-            xax_3hr=np.array([3.0*x-6.5 for x in range(8)])
-            xax_1hr=np.array([1.0*x-6.5 for x in range(24)]) 
+            ax  =fig.add_axes([0.10, 0.10, 0.85, 0.85]) # Create axes
+            #-----------------------------------------------------------------------------
+            # create site-dependent time offset for the local time & plotting range [XZ]
```

Comment:
Thank you for fixing the site depended local time offset!

Reply:
You're welcome!